lang/cs/Org.Apache.REEF.Client/AzureBatch/Service/AzureBatchService.cs

```diff
@@ -28,32 +29,44 @@
 
 namespace Org.Apache.REEF.Client.DotNet.AzureBatch
 {
-    public sealed class AzureBatchService : IDisposable
+    internal sealed class AzureBatchService : IDisposable
     {
         private static readonly Logger LOGGER = Logger.GetLogger(typeof(AzureBatchService));
         private static readonly TimeSpan RetryDeltaBackOff = TimeSpan.FromSeconds(5);
         private const string AzureStorageContainerSasToken = "AZURE_STORAGE_CONTAINER_SAS_TOKEN_ENV";
         private const int MaxRetries = 3;
 
         public BatchSharedKeyCredential Credentials { get; private set; }
-        public string PoolId { get; private set; }
+        public string PoolId { get; }
+
+        private BatchClient Client { get; }
+        private ContainerRegistryProvider ContainerRegistryProvider { get; }
+        private IList<string> Ports { get; }
+        private ICommandBuilder CommandBuilder { get; }
+        private bool AreContainersEnabled => this.ContainerRegistryProvider.IsValid();
 
-        private BatchClient Client { get; set; }
         private bool disposed;
 
         [Inject]
         public AzureBatchService(
+            ContainerRegistryProvider containerRegistryProvider,
+            ICommandBuilder commandBuilder,
             [Parameter(typeof(AzureBatchAccountUri))] string azureBatchAccountUri,
             [Parameter(typeof(AzureBatchAccountName))] string azureBatchAccountName,
             [Parameter(typeof(AzureBatchAccountKey))] string azureBatchAccountKey,
-            [Parameter(typeof(AzureBatchPoolId))] string azureBatchPoolId)
+            [Parameter(typeof(AzureBatchPoolId))] string azureBatchPoolId,
+            [Parameter(typeof(AzureBatchPoolDriverPortsList))] IList<string> ports)
         {
-            BatchSharedKeyCredential credentials = new BatchSharedKeyCredential(azureBatchAccountUri, azureBatchAccountName, azureBatchAccountKey);
+            BatchSharedKeyCredential credentials =
+                new BatchSharedKeyCredential(azureBatchAccountUri, azureBatchAccountName, azureBatchAccountKey);
 
+            this.Ports = ports;
```

Comment:
Why use Java practices like 'this' as opposed to C# practices (i.e., '_') for referencing instance variables?

Reply:
I would follow a specific standard if there was one. I find cases where there are classes with _memberVariables as as Properties. For example, look at reef\lang\cs\Org.Apache.REEF.Tang\Annotations\NamedParameter.cs and the code in this class is not much different from that one.

That said, if there is a steady effort to move towards one notation and away from a different one, I am willing to comply. As it stands now, I see multiple "standards" and I can only guess which one is the right one.

Won't fix for now.

---
In reply to: [206968926](https://github.com/apache/reef/pull/1478#discussion_r206968926) [](ancestors = 206968926)

Reply:
The coding standards for REEF.NET point to the coding guidelines followed by .NET Core open source (see https://cwiki.apache.org/confluence/display/REEF/Coding+Guidelines) 

There (https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) under the third item you will see that you should "Prefix internal and private instance fields with _"

It's unfortunate that prior merges did not address this situation, but if it's not too much trouble, then it would be good to move the needle in the recommended direction.

Reply:
Item 4 is also fairly explicit about the use of "this."